### PR TITLE
Week 5 octavio sifuentes

### PR DESCRIPTION
**Pull request by priceymineral**
# Submitting Week 5 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [✅] JavaScript Coding Challenges
- [✅] Ruby Coding Challenges
- [✅] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: ruby-challenges.rb [hunk 1]
```diff
@@ -10,23 +10,113 @@
...
+# p my_bike
+# p my_bike.bike_info
 
 # -------------------3b) Add the ability to pedal faster and brake. The pedal_faster method should increase the speed by a given amount. The brake method should decrease the speed by a given amount. The bike cannot go negative speeds.
 
 # Expected output example: my_bike.pedal_faster(10) => 10
 # Expected output example: my_bike.pedal_faster(18) => 28
 # Expected output example: my_bike.brake(5) => 23
 # Expected output example: my_bike.brake(25) => 0
+
```
Comment (TsunamiCode5) [inline ruby-challenges.rb:91]: Very well done.

File: ruby-challenges.rb [hunk 2]
```diff
@@ -10,23 +10,113 @@
...
+def hash_values_in_order(hash) 
+    values = []
+    hash.each do |key, value|
+        values.push(value)
+    end
+    values.flatten.sort
+end
+
+p hash_values_in_order(us_states)
```
Comment (TsunamiCode5) [inline ruby-challenges.rb:59]: 😍

File: ruby-challenges.rb [hunk 3]
```diff
@@ -10,23 +10,113 @@
 letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+# PSEUDO
```
Comment (TsunamiCode5) [inline ruby-challenges.rb:13]: AWESOME!

File: interview-questions.md [hunk 4]
```diff
@@ -8,44 +8,58 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-Researched answer:
+Researched answer: exmaples of gems are: Rails, RSpec, Bundler, etc.
 
 3. What is a relational database? Are there other kinds of databases?
 
-Your answer:
+Your answer: A relational database consists of tables with colums and values. Tables can be related via the use of foreign keys. Relational databases are typically accessed via SQL commands. There is also nonrelationl databases which use documents which function more like an object and can be interacted without SQL. 
 
-Researched answer:
+Researched answer: Relational databases hold collections of data organized in tables. The use of tables makes it easy to see the relatonships between different data within the collection, whether that relationship is within the table or with data from other tables. Relational databases use Structured Query Language (SQL). Other types of databases are:
```
Comment (TsunamiCode5) [inline interview-questions.md:26]: Hired!

File: code-challenges.test.js [hunk 5]
```diff
@@ -50,3 +125,75 @@ const hand4 = [7, 2, 7, 2, 7]
...
+            if (++quit > 2) return false
+            counts[card] = 1
+        }
+    }
+
+    if (Object.values(counts).includes(2)) return true
+
+    return false
+}
+// Attempt to do challenge 3 using Map()
```
Comment (TsunamiCode5) [inline code-challenges.test.js:173]: This is the best thing that I got in the habit of doing: documenting the process and trials. These assessments are for you to have access to always, so you can always look back at the different approaches you have tried before. Great notes for yourself in the future.

**Review by TsunamiCode5 — APPROVED** [review]
Octavio, great job all around on this assessment. You are showing so much growth as a developer. Every assessment you turn in gets even better, and you are more confident each week to answer during lectures, or help talk things through with your cohort mates. This assessment comprehension level is competent. Keep powering through!

**Comment by priceymineral** [comment]
> Octavio, great job all around on this assessment. You are showing so much growth as a developer. Every assessment you turn in gets even better, and you are more confident each week to answer during lectures, or help talk things through with your cohort mates. This assessment comprehension level is competent. Keep powering through!

**Pull request merged**